_posts/2020-10-31-monitoring-spring-prometheus-grafana.md

```diff
@@ -0,0 +1,342 @@
+---
+layout: post
+authors: [kevin_govaerts]
+title: 'Monitoring Spring boot with Prometheus and Grafana'
+image: /img/2020-10-31-monitoring-spring-prometheus-grafana/thumbnail.jpg
+tags: [Spring, Prometheus, Grafana, Docker]
+category: Monitoring
+comments: true
+---
+
+# Table of contents
+{:.no_toc}
+
+- TOC
+{:toc}
+
+----
+
+# Introduction
+
+In a distributed landscape where we are working with microservices, serverless applications or just event-driven architecture as a whole, than observability, which comprises monitoring, logging, tracing and alerting, is an important architectural concern.  
+
+There are a few reasons why we want visibility in our highly distributed systems:  
+- Issues will occur, even when our best employees have built it.
+- Distributed systems generate distributed failures, which can be devastating when we are not prepared in advance.
+- Reveal mistakes early, which is great for improvement and learning.
+- It keeps us accountable.
+- Reduce the mean time to resolution (MTTR).
+
+In this blogpost I will explain the core concepts of Prometheus and Grafana.  
+In the last section I set up a demo project, so you can follow along and implement monitoring in     your own applications. 
+
+# Prometheus
+
+## What is Prometheus? 
+
+Prometheus is an open source monitoring system, originally developed by SoundCloud, it  has a large community behind it and can monitor nearly anything.  
```

Comment:
```suggestion
Prometheus, originally developed by SoundCloud is an open source and community-driven project that graduated from the Cloud Native Computing Foundation.
It can aggregate data from almost everything:
```